atom/browser/browser_mac.mm

```diff
@@ -193,19 +194,27 @@
 Browser::LoginItemSettings Browser::GetLoginItemSettings(
     const LoginItemSettings& options) {
   LoginItemSettings settings;
+#if defined(MAS_BUILD)
+  settings.open_at_login = platform_util::GetLoginItemEnabled();
+#else
```

Comment:
As far as I know there's no good way of deciding this. I just set the item and check whether the call returns `true`.